gossip/src/crds_value.rs

```diff
@@ -485,6 +498,69 @@ impl Sanitize for NodeInstance {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Default, PartialEq, Eq, AbiExample)]
+pub struct RestartLastVotedForkSlots {
+    pub slots: EpochSlots,
```

Comment:
Why are you using `EpochSlots` here?
How are the `EpochSlots` related to `RestartLastVotedForkSlots`?

Reply:
I mainly want to reuse the compress/uncompress logic of EpochSlots, all I need is adding last_vote to EpochSlots.

Reply:
I think you want to use something like `Vec<CompressedSlots>`:
https://github.com/solana-labs/solana/blob/5dbc19ccb/gossip/src/epoch_slots.rs#L230
not the entire `EpcohSlots`.

Reply:
Also why do we need `last_voted_slot` separately?
Isn't that already the last slot in the `slots` field?

Reply:
Because one Gossip message may not fit 81k slots, so we need to split it into multiple RestartLastVotedForkSlots messages, only the last message has last_voted_slot.

Reply:
If I get one of those `RestartLastVotedForkSlots` messages before the last one which says, for example:
```rust
last_voted_slot == 1000
slots.last() == 10
```
how are we going to identify that `10` is not the parent of `10000`? We may not ever get the last `RestartLastVotedForkSlots` for that node.

Reply:
Right now the protocol only does pure aggregation, when you get RestartLastVotedForkSlots from any staked nodes it means that person has voted for all the slots in this message. last_voted_slot and last_voted_hash and timestamp are used together with pubkey to be unique key, in case some validator sends out a new message and we want to replace previous votes.

We can tolerate about 5% packet loss, if we have more than 5% packet loss then this restart protocol might not give you a definite answer, but it has at least made an effort to replace some important blocks which >42% of the validators already voted for.

Reply:
Might need a counter on the last vote to tell people what the expected number of `RestartLastVotedForkSlots` is

Reply:
We won't misinterpret the parent, because when we traverse to select HeaviestFork we are using the parent information on the BankFork. But it's nice not to miss part of the RestartLastVotedForkSlots, because that might mean you didn't count all of the stakes. I'll add the counter.